src/MongoDB/BSON/ObjectID.cpp

```diff
@@ -35,8 +36,8 @@ void HHVM_METHOD(MongoDBBsonObjectID, __construct, const Variant &objectId)
 	const String& str_objectId = objectId.isNull() ? null_string : objectId.toString();
 
 	if (!objectId.isNull()) {
-		if (bson_oid_is_valid(str_objectId.c_str(), str_objectId.length())) {
-			bson_oid_init_from_string(&data->m_oid, str_objectId.c_str());
+		if (bson_oid_is_valid((HHVM_FN(strtolower)(str_objectId)).c_str(), str_objectId.length())) {
```

Comment:
In case you wonder whether this causes a memory leak. It doesn't, as strtolower modifies the existing string.